Fix URL encoding in content accessor by replacing quote_plus with quote for file URLs

pranshugupta01 opened the pull request:
## What type of PR is this? (Check all that apply)

- [ ] 🛠 Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] ⚡ Optimization
- [ ] 📝 Documentation Update

## Description
This PR fixes an issue with the file_download control when file names contain spaces. Previously, spaces in filenames were encoded as **`+`** characters (using **`urllib.parse.quote_plus`**), which caused browsers to fail to locate the files. This PR changes the URL encoding to use **`urllib.parse.quote`** instead, which properly encodes spaces as **`%20`** in URLs, allowing files with spaces in their names to be downloaded correctly.

## Related Tickets & Documents

- Closes #2523 

## How to reproduce the issue

1. Create two text files in the same directory: sample_file.txt and sample file.txt (with a space)
2. Create a Taipy app with: 
```
 import taipy.gui.builder as tgb
   from taipy import Gui

   with tgb.Page() as test_page:
       tgb.text("# Test Download button", mode="md")
       tgb.file_download(
           content="sample file.txt",  # Use file with space in name
           label="Download File",
       )

   if __name__ == "__main__":
       gui = Gui(page=test_page)
       gui.run(
           title="Test download",
           use_reloader=True,
       )
```

3. Run the app and click the download button
4. Before the fix: A .htm file is downloaded with a 404 error message
5. After the fix: The text file is properly downloaded

## Backporting
<!-- Specify any branches or releases this change needs to be backported to. -->
_This change should be backported to:_
- [ ] 3.0
- [ ] 3.1
- [ ] 4.0
- [ ] develop

## Checklist
_We encourage keeping the code coverage percentage at **80% or above**._
- [x] ✅ This solution meets the acceptance criteria of the related issue.
- [ ] 📝 The related issue checklist is completed.
- [x] 🧪 This PR includes **unit tests** for the developed code.
    If not, explain why:
- [ ] 🔄 **End-to-End tests** have been added or updated.
    If not, explain why:
- [ ] 📚 The **documentation** has been updated, or a dedicated issue has been created.
    If not, explain why:
- [ ] 📌 The **release notes** have been updated.
    If not, explain why:

pranshugupta01 committed:
Fix URL encoding in content accessor by replacing quote_plus with quote for file URLs

FredLL-Avaiga reviewed (changes requested):
Can you add a test ?

pranshugupta01 committed:
Add test for file download handling with spaces in file path

This commit introduces a new test case to verify the functionality of file downloads when the file path contains spaces. The test creates a temporary file, sets the content variable in the GUI, and checks the generated markdown string against expected output, ensuring proper URL encoding for spaces.

pranshugupta01 commented:
Hey @FredLL-Avaiga i have added test , please verify

FredLL-Avaiga commented on tests/gui/control/test_file_download.py:
```diff
@@ -50,6 +50,27 @@ def test_file_download_path_md(gui: Gui, test_client, helpers):
+        md_string = "<|{content}|file_download|>"
+        expected_list = [
+            "<FileDownload",
+            'defaultContent="/taipy-content/taipyStatic',
+            'test%20file%20with%20spaces.txt"',
```
Why not on the same line ?

pranshugupta01 replied:
done

pranshugupta01 committed:
Merge branch 'develop' into bug/#2523-file-download-with-spaces

pranshugupta01 committed:
Update expected output for file download test to reflect correct file path with spaces

This commit modifies the expected output in the test for file downloads to ensure the correct URL is generated when the file path contains spaces. The change updates the defaultContent attribute to include the proper path format.

pranshugupta01 committed:
Merge branch 'bug/#2523-file-download-with-spaces' of github.com-pranshugupta01:pranshugupta01/taipy into bug/#2523-file-download-with-spaces

FredLL-Avaiga reviewed (approved):
Excellent, tx a lot @pranshugupta01
